File: services/agent-memory/internal/repoindexer/ast/testdata/polyglot/hello.cpp
// Polyglot smoke fixture: C++.
// Must declare a class/type, a free function, a same-file call,
// and one import so the dispatcher emits >=1 class node,
// >=1 method node, and >=1 static_calls edge per language.
// The class counts as the type; the same-file call flows from
// greet -> format_greeting (both translation-unit free funcs).
#include <string>

class Greeter {
public:
  int n;
};

int format_greeting(int n) {
  return n + 1;
}

int greet(int n) {
  return format_greeting(n);
}

File: services/agent-memory/internal/repoindexer/ast/testdata/polyglot/hello.go
// Polyglot smoke fixture: Go.
// Must declare a class/type, a free function, a same-file call,
// and one import so the dispatcher emits >=1 class node,
// >=1 method node, and >=1 static_calls edge per language.
// The struct counts as the type; the same-file call flows
// from greet -> formatGreeting (both package-level free funcs).
package hello

import "fmt"

type Greeter struct {
	prefix string
}

func formatGreeting(name string) string {
	return fmt.Sprintf("hi %s", name)
}

func greet(name string) string {
	return formatGreeting(name)
}
